### ISSUE TITLE
Enable warnings

### ISSUE THREAD

**Issue by knatten** [zivid/cpp-extra-samples#36]
We currently build the code without warnings. We should enable as many warnings as possible, and build with warnings as errors in CI. I suggest not having warnings as errors outside CI, as that will make it harder for users to experiment with the samples.

**Comment by knatten** [comment]
I've started on this in the [warnings](https://github.com/zivid/cpp-extra-samples/tree/warnings) branch.

**Comment by knatten** [comment]
We're currently only using Clang 8 in CI. If don't have Clang 8, and see some warnings from it in CI which you can't reproduce with your local compiler, you can run the CI environment locally:

Start Docker:
```
docker run -it --volume $PWD:/host --workdir /host/continuous-integration/linux ubuntu:18.04 bash
```

Run setup:
```
./setup.sh
```

Run CI build:
```
./build.sh
```

**Comment by knatten** [comment]
Feedback from Mathias, don't add even add the warnings outside of CI. Look at the Python repo for inspiration.

**Comment by knatten** [comment]
Rebased the warnings branch on master now

**Comment by knatten** [comment]
Have merged #39 now, and I'm closing this issue. Will continue in #41 .
